pkg/oc/cli/admin/release/extract_tools.go

```diff
@@ -125,13 +132,50 @@ func (o *ExtractOptions) extractCommand(command string) error {
 			return fmt.Errorf("the supported commands are 'oc' and 'openshift-install'")
 		}
 	} else {
+		willArchive = true
 		targets = availableTargets
 		for i := range targets {
 			targets[i].AsArchive = true
 			targets[i].AsZip = targets[i].OS == "windows"
 		}
 	}
 
+	var hashFn = sha256.New
+	var signer *openpgp.Entity
+	if willArchive && len(o.SigningKey) > 0 {
+		key, err := ioutil.ReadFile(o.SigningKey)
+		if err != nil {
+			return err
+		}
+		keyring, err := openpgp.ReadArmoredKeyRing(bytes.NewBuffer(key))
+		if err != nil {
+			return err
+		}
+		for _, key := range keyring {
+			if !key.PrivateKey.CanSign() {
+				continue
+			}
+			fmt.Fprintf(o.Out, "Enter password for private key: ")
+			password, err := terminal.ReadPassword(int(syscall.Stdin))
+			fmt.Fprintln(o.Out)
+			if err != nil {
+				return err
+			}
+			if err := key.PrivateKey.Decrypt(password); err != nil {
+				return fmt.Errorf("unable to decrypt signing key: %v", err)
+			}
+			for i, subkey := range key.Subkeys {
+				if err := subkey.PrivateKey.Decrypt(password); err != nil {
+					return fmt.Errorf("unable to decrypt signing subkey %d: %v", i, err)
+				}
+			}
+			signer = key
```

Comment:
`break`?

Reply:
Not having a `break` is not really a big deal, this UX isn't set up for keyfiles that contain multiple signing keys anyway, and it's probably not worth another round of CI ;).